service/history/workflow/workflow_task_state_machine.go

```diff
@@ -1047,7 +1053,9 @@ func cleanTaskQueue(tq *taskqueuepb.TaskQueue) *taskqueuepb.TaskQueue {
 	if err != nil {
 		return tq
 	}
-	cleanTq := *tq
+
+	// FIXME: shallow clone
```

Comment:
I deleted all these FIXMEs and we're just going to clone these unless it shows up in a performance profile. They were done not for performance reasons but to prevent alterations from affecting the source object

Reply:
Yes, I know why we have it. I did most of them :-). Ok, I generally agree with approach. Although there are few on a very hot path.